metadata-ingestion/src/datahub/ingestion/source/salesforce.py

```diff
@@ -0,0 +1,737 @@
+
+    def __init__(self, config: SalesforceConfig, ctx: PipelineContext) -> None:
+        super().__init__(ctx)
+        self.config = config
+        self.report = SalesforceSourceReport()
+        self.session = requests.Session()
+        self.platform: str = "salesforce"
+        self.fieldCounts = {}
+
+        try:
+            if self.config.auth is SalesforceAuthType.DIRECT_ACCESS_TOKEN:
+                logger.debug("Access Token Provided in Config")
+                assert (
+                    self.config.access_token is not None
+                ), "Config access_token is required for DIRECT_ACCESS_TOKEN auth"
+                assert (
+                    self.config.instance_url is not None
+                ), "Config instance_url is required for DIRECT_ACCESS_TOKEN auth"
+
+                self.sf = Salesforce(
+                    instance_url=self.config.instance_url,
+                    session_id=self.config.access_token,
+                    session=self.session,
+                )
+            elif self.config.auth is SalesforceAuthType.USERNAME_PASSWORD:
+                logger.debug("Username/Password Provided in Config")
+                assert (
+                    self.config.username is not None
+                ), "Config username is required for USERNAME_PASSWORD auth"
+                assert (
+                    self.config.password is not None
+                ), "Config password is required for USERNAME_PASSWORD auth"
+                assert (
+                    self.config.security_token is not None
+                ), "Config security_token is required for USERNAME_PASSWORD auth"
+
+                self.sf = Salesforce(
+                    username=self.config.username,
+                    password=self.config.password,
+                    security_token=self.config.security_token,
+                    session=self.session,
+                )
+
+        except Exception as e:
+            logger.error(e)
+            raise ConfigurationError("Salesforce login failed")
+        else:
+            # List all REST API versions and use latest one
+            versions_url = "https://{instance}/services/data/".format(
+                instance=self.sf.sf_instance,
+            )
+            versions_response = self.sf._call_salesforce("GET", versions_url).json()
+            latest_version = versions_response[-1]
+            version = latest_version["version"]
+            self.sf.sf_version = version
+
+            self.base_url = "https://{instance}/services/data/v{sf_version}/".format(
+                instance=self.sf.sf_instance, sf_version=version
+            )
+
+            logger.debug(
+                "Using Salesforce REST API with {label} version: {version}".format(
+                    label=latest_version["label"], version=latest_version["version"]
+                )
+            )
+
+    @classmethod
+    def create(cls, config_dict: dict, ctx: PipelineContext) -> "Source":
```

Comment:
you no longer need to define the `create` method if you have annotated your class with `@config_class`